conans/test/functional/configuration/profile_test.py

```diff
@@ -635,3 +635,13 @@ def test_profile_crazy_inheritance(self):
                              compiler.runtime=MD
                              compiler.version=15
                              os=Windows"""), self.client.out)
+
+
+def test_profile_from_cache_path():
```

Comment:
@uilianries please check this new test that I have provided. This would be proving that the previous fix was breaking, very likely massively breaking many users (all of them that manages profiles in subfolders in the cache).

The new proposed fix is more conservative and passes your new tests, please check if there is some other case to take into account.

Reply:
I would love adding a functional test to validate relative paths too. I know those unit tests cover very well `get_profile_path()`, but I would like having and end-to-end, because we have a considerable stack before calling it.

![stacktrace](https://user-images.githubusercontent.com/4870173/113623139-509f1200-9634-11eb-9ea3-17fda27fc68e.png)


For regression it can help us.

Reply:
Actually this should go to ``integration``, not functional (as they don't depend on external tools), but until this is fixed, yes, it can co here.

Yes, it makes sense to have a few more tests here to cover better this functionality, please add some. Try to keep them simple, as the proposed one. Thanks!